pkg/genericapiserver/api/handlers/negotiation/negotiate.go

```diff
@@ -66,7 +66,7 @@ func NegotiateOutputStreamSerializer(req *http.Request, ns runtime.NegotiatedSer
 func NegotiateInputSerializer(req *http.Request, ns runtime.NegotiatedSerializer) (runtime.SerializerInfo, error) {
 	mediaTypes := ns.SupportedMediaTypes()
 	mediaType := req.Header.Get("Content-Type")
-	if len(mediaType) == 0 {
+	if len(mediaType) == 0 && len(mediaTypes) > 0 {
```

Comment:
It should be a construction-time error to make a serializer without any mediaTypes, I think. Silently failing here is probably not great. @smarterclayton can confirm.